frontend/src/metabase/query_builder/actions/ui.ts

```diff
@@ -61,10 +62,10 @@ export const onCloseSidebars = createAction("metabase/qb/CLOSE_SIDEBARS");
 
 export const CLOSE_QB_NEWB_MODAL = "metabase/qb/CLOSE_QB_NEWB_MODAL";
 export const closeQbNewbModal = createThunkAction(CLOSE_QB_NEWB_MODAL, () => {
-  return async (dispatch, getState) => {
+  return async (_dispatch, getState) => {
     // persist the fact that this user has seen the NewbModal
     const { currentUser } = getState();
-    await UserApi.update_qbnewb({ id: currentUser.id });
+    await UserApi.update_qbnewb({ id: currentUser?.id });
```

Comment:
Current user can be `null`.

Reply:
That is a good question.
I don't know why the currentUser type was defined as potentially `null` in
https://github.com/metabase/metabase/blob/master/frontend/src/metabase-types/store/state.ts

Maybe @kulyk knows?

The type might refer to the fact that a user can be logged out, perhaps?

Reply:
I tried removing null, and there are only 2 places that need fixing in unit tests.

I don't think there's a scenario for the `currentUser` to be null here, since JS code always assumes it's not null and that would have crashed already if it could be null. That means this looks like a typing problem.

Reply:
I think I found an answer
https://github.com/metabase/metabase/pull/26477#pullrequestreview-1179305195

Reply:
I'm not sure what should we do in this case then. It's true that `User | null` reflects the actual behavior of the state, but 99% of the time, we would deal with the code that is already for [authenticated users.](https://github.com/metabase/metabase/blob/f016befaefc930fab198ce57c05938b3c269a321/frontend/src/metabase/routes.jsx#L124) In that sense, when you call `getUser` inside that block, `currentUser` shouldn't be null.

One way to get out of this is to specifically cast `currentUser` to `User` until we can figure out a better solution for `currentUser` state type. That we we're explicit about expecting that `currentUser` shouldn't be `null` at this point.

Reply:
@ranquild can you chime in here, please?

Reply:
I think the types should be 100% correct. If it can be `null`, it should be `null`. You can introduce a helper selector that gets the user and throws if it's null.

Reply:
Totally forgot we have `checkNotNull` for this very use case.
https://github.com/metabase/metabase/blob/f016befaefc930fab198ce57c05938b3c269a321/frontend/src/metabase/lib/types.ts#L15

Although, that still means `currentUser` would still be `User | null`. I think we deal with the nullable values this way most of the time.

Reply:
Yup.
https://github.com/metabase/metabase/blob/master/frontend/src/metabase/account/password/containers/UserPasswordApp/UserPasswordApp.tsx#L11